src/budget/CategoryRow.js

```diff
@@ -1,15 +1,129 @@
 import React from 'react';
+import { PropTypes } from 'prop-types';
+import { useSelector } from 'react-redux';
+import Button from 'react-bootstrap/Button';
+import ProgressBar from 'react-bootstrap/ProgressBar';
 
 /**
  * Shows the category title, a progress bar, how much was spent out of the limit
  * and a button to view more details
+ * expects to go in a thing with a variable of whatever
  *
  * @component
  * @example
  *
+ * PUT AN EXAMPLE HERE
  */
-export default function CategoryRow() {
+
+// Need to do:
+// send data in details link
+//  ?what happens on the details page?
+//   <Route path="budget/:categoryId" element={<CategoryDetailsPage />} />
+// ?how should I be getting the time period?
+//   Hard code for now until there's a means to get the desired period.
+// test edge cases
+// edge cases:
+// underspent (return): $0 out of $50 (crossed out) $90 with empty bar
+// overspent: $120 out of $100 with full bar
+
+function CategoryRow({ category }) {
+  const startDate = new Date('06-01-2021');
+  const endDate = new Date('06-30-2021');
+  // check out memoization
+  const amountSpent = useSelector((state) => state.transactions.entities.filter(
+    (transaction) => transaction.categoryId === category.id
+      && transaction.date >= startDate && transaction.date <= endDate,
+  ).reduce((accumulator, currentValue) => accumulator + currentValue.amount, 0).toFixed(2)) * -1;
+
```

Comment:
Very tiny thing, let's move that * -1 inside the call to useSelector(). So just inside the outermost parenthesis.

We want all our data manipulation to happen inside useSelector because it keeps everything organized and it allows us to optimize nicely later.